mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -258,6 +290,27 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
+  @Since("2.0.0")
+  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
```

Comment:
The way this PR is currently, you have to set initial model AND set k. I think that's annoying. This should work:

````scala
val kmeansModel = new KMeans().setInitialModel(model).fit(df)
````

If a user sets the initial model and explicitly sets k to some conflicting value, I think we should throw an error. However, if they don't set anything for k, we should just use the initial model k. Currently, k defaults to 2 so an error gets thrown (unless the initial model has 2 centers).